lib/chat_statistics.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class DiscourseChat::Statistics
+  def self.about_messages
+    {
+      last_day: ChatMessage.where('created_at > ?', 1.days.ago).count,
+      "7_days" => ChatMessage.where('created_at > ?', 7.days.ago).count,
+      "30_days" => ChatMessage.where('created_at > ?', 30.days.ago).count,
+      previous_30_days: ChatMessage.where('created_at BETWEEN ? AND ?', 60.days.ago, 30.days.ago).count,
```

Comment:
@martin-brennan I don't think we have a index that covers these queries here. The queries are likely going to be expensive because the `chat_messages` table is likely to have alot of rows.

Reply:
Oh yeah good idea, we have that on `Post` and `Topic` in core which are used for `About` there. We don't have it on `UserAction` (for likes) or `User` though, core stats here:

https://github.com/discourse/discourse/blob/98bacbd2c6b9fe57167cd32af5eb4839b4a5d1f6/app/models/about.rb#L82-L104